src/RamStatement.h

```diff
@@ -446,27 +425,27 @@ class RamSequence : public RamListStatement {
  */
 class RamParallel : public RamListStatement {
 public:
+    RamParallel(std::vector<std::unique_ptr<RamStatement>> statements)
+            : RamListStatement(std::move(statements)) {}
     RamParallel() : RamListStatement() {}
-
-    void print(std::ostream& os, int tabpos) const override {
-        os << times(" ", tabpos) << "PARALLEL" << std::endl;
-        for (auto const& stmt : statements) {
-            stmt->print(os, tabpos + 1);
-        }
-        os << times(" ", tabpos) << "END PARALLEL" << std::endl;
-    }
+    template <typename... Stmts>
+    RamParallel(std::unique_ptr<RamStatement> f, std::unique_ptr<Stmts>... rest)
```

Comment:
`f` is not a very descriptive term for what this is
```suggestion
    RamParallel(std::unique_ptr<RamStatement> first, std::unique_ptr<Stmts>... rest)
```